src/main/java/com/google/finapp/DatabaseModule.java

```diff
@@ -28,21 +28,25 @@
 final class DatabaseModule extends AbstractModule {
 
   @Override
-  protected void configure() {
-    bind(SpannerDaoInterface.class).to(SpannerDaoImpl.class);
-  }
+  protected void configure() {}
 
   @Provides
   @Singleton
-  DatabaseClient provideDatabaseClient(
+  SpannerDaoInterface provideSpannerDao(
       @ArgsModule.SpannerHost String spannerHost,
       @ArgsModule.SpannerPort int spannerPort,
       @ArgsModule.SpannerProjectId String spannerProjectId,
       @ArgsModule.SpannerInstanceId String spannerInstanceId,
-      @ArgsModule.SpannerDatabaseId String spannerDatabaseId) {
+      @ArgsModule.SpannerDatabaseId String spannerDatabaseId,
+      @ArgsModule.SpannerDaoImpl String spannerDaoImpl) {
     SpannerOptions spannerOptions = SpannerOptions.getDefaultInstance();
     Spanner spanner = spannerOptions.toBuilder().build().getService();
-    return spanner.getDatabaseClient(
+    DatabaseClient client = spanner.getDatabaseClient(
         DatabaseId.of(spannerProjectId, spannerInstanceId, spannerDatabaseId));
+    if (spannerDaoImpl == "jdbc") {
+      return new SpannerDaoJdbcImpl(client);
+    } else {
```

Comment:
It would be preferable behavior to check for some other string e.g: "client_lib".

Here's one reason, assume the customer mis-spelled, or capitalized, e.g: JDBC, the result would be to run with client library, but it would be preferable to throw an exception with the supported flag values.

Reply:
Look at my other comment about using a boolean instead. That's my opinion, at least.